ingestclient/plugins/stack.py

```diff
@@ -142,23 +144,29 @@ def process(self, file_path, x_index, y_index, z_index, t_index=0):
         """
         # Load tile
         file_handle = self.fs.get_file(file_path)
-
         x_range = [self.parameters["ingest_job"]["tile_size"]["x"] * x_index,
                    self.parameters["ingest_job"]["tile_size"]["x"] * (x_index + 1)]
         y_range = [self.parameters["ingest_job"]["tile_size"]["y"] * y_index,
                    self.parameters["ingest_job"]["tile_size"]["y"] * (y_index + 1)]
 
-        # Allowed data types:
-        allowed_types = [np.uint8, np.uint16, np.uint64]
+        tile_crop_index_left = x_range[0] - self.parameters["ingest_job"]["extent"]["x"][0] 
+        tile_crop_index_right =  tile_crop_index_left + self.parameters["ingest_job"]["tile_size"]["x"]
+        tile_crop_index_upper = y_range[0] - self.parameters["ingest_job"]["extent"]["y"][0]
+        tile_crop_index_lower = tile_crop_index_upper + self.parameters["ingest_job"]["tile_size"]["y"]
 
         # Save sub-img to png and return handle
-        tile_data = Image.open(file_handle)
-        tile_arr = np.array(tile_data)
-        if tile_arr.dtype not in allowed_types:
-            print("Your data type is not uint8, uint16 or uint64, converting to uint8 and attempting upload.")
-            tile_arr = np.uint8(tile_arr/256)
-            tile_data = Image.fromarray(tile_arr)
-        upload_img = tile_data.crop((x_range[0], y_range[0], x_range[1], y_range[1]))
+        try:
+            tile_data = Image.open(file_handle)
+            tile_arr = np.array(tile_data)
+            if tile_arr.dtype not in allowed_types:
+                print("Your data type is not uint8, uint16 or uint64, converting to uint8 and attempting upload.")
+                tile_arr = np.uint8(tile_arr/256)
+                tile_data = Image.fromarray(tile_arr)
+            upload_img = tile_data.crop((tile_crop_index_left, tile_crop_index_upper, tile_crop_index_right, tile_crop_index_lower))
+        except OSError:
```

Comment:
Should print out what the specific `OSError` is, also.

Reply:
Yeah, I might also make some changes to the code.... This PR was a bit premature. Thanks for the feedback!